hack/update-etcd-objects.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to update etcd objects as per the latest API Version.
+# This just reads all objects and then writes them back as is to ensure that
+# they are written using the latest API version.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBECTL="./cluster/kubectl.sh"
+
+# List of resources to be updated.
+declare -a resources=(
+    "endpoints"
+    "events"
+    "limitranges"
+    "namespaces"
+    "nodes"
+    "pods"
+    "persistentvolumes"
+    "persistentvolumeclaims"
+    "rc"
+    "resourcequotas"
+    "secrets"
+    "services"
+)
+
+for resource in "${resources[@]}"
+do
+  instances=( $(${KUBECTL} get $resource -o template -t "{{range.items}}{{.metadata.name}} {{end}}"))
```

Comment:
https://github.com/GoogleCloudPlatform/kubernetes/pull/7024

----- Original Message -----
> > +    "events"
> > +    "limitranges"
> > +    "namespaces"
> > +    "nodes"
> > +    "pods"
> > +    "persistentvolumes"
> > +    "persistentvolumeclaims"
> > +    "rc"
> > +    "resourcequotas"
> > +    "secrets"
> > +    "services"
> > +)
> > +
> > +for resource in "${resources[@]}"
> > +do
> > +  instances=( $(${KUBECTL} get $resource -o template -t
> > "{{range.items}}{{.metadata.name}} {{end}}"))
> 
> Thats a great point @lavalamp. This works on objects in the default namespace
> only. Thanks for pointing it out.
> I can first fetch all the namespaces and then do this for all of them.
> But it will be a lot easier, if we have plans to add --all-namespaces flag.
> @smarterclayton Can you please point me to the issue where it is proposed?
> Is it https://github.com/GoogleCloudPlatform/kubernetes/issues/7793?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7819/files#r29739216